tls/s2n_connection.h

```diff
@@ -199,6 +199,9 @@ struct s2n_connection {
      */
     uint8_t resume_protocol_version;
 
+    /* The protocol version in the record header of the client hello message */
+    uint8_t client_hello_record_version;
+
```

Comment:
Should this be on the connection or on the ClientHello?

Our current ClientHello parsing method only takes the message itself, but that might not always be true. We could write a version that takes a record / raw TCP data.